torchvision/utils.py

```diff
@@ -382,6 +389,116 @@ def draw_keypoints(
     return torch.from_numpy(np.array(img_to_draw)).permute(2, 0, 1).to(dtype=torch.uint8)
 
 
+# Flow visualization code adapted from https://github.com/tomrunia/OpticalFlow_Visualization
+@torch.no_grad()
+def flow_to_image(
+    flow: torch.Tensor,
+) -> torch.Tensor:
+
+    """
+    Converts the two-dimensional flow to an RGB Image.
+
+    Args:
+        flow (Tensor): Flow of shape (2, H, W) and dtype torch.float.
+
+    Returns:
+        img (Tensor(3, H, W)): Image Tensor where each color corresponds to a given flow direction.
+    """
+
+    if flow.ndim != 3 or flow.size(0) != 2:
+        raise ValueError(f"Input flow should have shape (2, H, W), got {flow.shape}.")
+
+    max_norm = torch.sum(flow ** 2, dim=0).sqrt().max()
+    epsilon = torch.finfo((flow).dtype).eps
+    normalized_flow = flow / (max_norm + epsilon)
+    return _normalized_flow_to_colors(normalized_flow)
+
+
+@torch.no_grad()
+def _normalized_flow_to_colors(normalized_flow: torch.Tensor) -> torch.Tensor:
+
+    """
+    Applies the flow color wheel to (possibly clipped) flow components u and v.
+
+    Args:
+        normalized_flow (torch.Tensor): Normalized flow tensor of shape (2, H, W)
+    Returns:
+       img (Tensor(3, H, W)): Flow visualization image of dtype uint8.
+    """
+
+    u = normalized_flow[0, :, :]
+    v = normalized_flow[1, :, :]
```

Comment:
Would it be possible to drop the use of `u` and `v` here as well?

Reply:
Can do that, but this code seems a bit more easy to read.

Reply:
I'm not sure I agree here :)

For example this would make it more obvious that we're computing the norm in the same way as above in the other function, and perhaps it would make https://github.com/pytorch/vision/pull/5134/files#r786121504 more obvious.

Also this is not "idiomatic" in the sense that this is not something that we've been using so far, so sticking with the full tensor would be more consistent in terms of style.